### ISSUE TITLE
v1.9.0: Possible false-positive for E3003

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#3541]
### CloudFormation Lint Version

cfn-lint 1.9.0

### What operating system are you using?

Windows 11

### Describe the bug

We get the following error:

`[cfn-lint] E3003: When using an ECS task definition of host port 0 and associating that container to an ELB the target group has to have a 'HealthCheckPort' of 'traffic-port'`

For 

```yaml
 TargetGroup:
    Type: AWS::ElasticLoadBalancingV2::TargetGroup
    Properties:
      VpcId: <VPCID>
      Port: 15671
      Protocol: TCP
      TargetType: ip
      TargetGroupAttributes:
        - Key: preserve_client_ip.enabled
          Value: true
        - Key: deregistration_delay.connection_termination.enabled
          Value: true
      HealthCheckEnabled: true
```

But we are able to deploy the following Stack in `eu-central-1`:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test


Resources:
  NLB:
    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
    Properties:
      IpAddressType: ipv4
      Scheme: internal
      Type: network
      SubnetMappings:
        - SubnetId: <SubnetID>


  Listener:
    Type: AWS::ElasticLoadBalancingV2::Listener
    Properties:
      LoadBalancerArn: !GetAtt NLB.LoadBalancerArn
      Port: 5671
      Protocol: TCP
      DefaultActions:
        - Type: forward
          TargetGroupArn: !GetAtt TargetGroup.TargetGroupArn


  TargetGroup:
    Type: AWS::ElasticLoadBalancingV2::TargetGroup
    Properties:
      VpcId: <VPCID>
      Port: 5671
      Protocol: TCP
      TargetType: ip
      TargetGroupAttributes:
        - Key: preserve_client_ip.enabled
          Value: true
        - Key: deregistration_delay.connection_termination.enabled
          Value: true
      HealthCheckEnabled: true
```

### Expected behavior

cfn-lint doesn't give an error.

### Reproduction template

See above. Not sure if the ECS Task Definition is missing in that example though.

**Comment by kddejong** [comment]
This rule is triggered from an AWS::ECS::Service resource and also draws in relationships from the associated ECS task definition.  I can't reproduce the issue with the resources above.  Any chance you can provide what you are using for those two resources?

**Comment by kddejong** [comment]
#3543 put one fix in but not sure it will fix this issue yet. My guess is it will.  I think the biggest thing I'm looking at is the PortMappings section for the appropriate `AWS::ECS::TaskDefinition`

**Comment by r-heimann** [comment]
> This rule is triggered from an AWS::ECS::Service resource and also draws in relationships from the associated ECS task definition. I can't reproduce the issue with the resources above. Any chance you can provide what you are using for those two resources?

I am unable to reproduce the error in v1.9.1, here is a very similar template that might have triggered the original issue:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test


Resources:
  KMSKey:
    Type: AWS::KMS::Key
    Properties:
      KeyPolicy:
        Version: 2012-10-17
        Id: !Sub "${AWS::AccountId}-${AWS::StackName}"
        Statement:
          - Sid: Allow root
            Effect: Allow
            Action: kms:*
            Resource: "*"
            Principal:
              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
          - Sid: "ECS: Allow generate data key access for Fargate tasks."
            Effect: Allow
            Principal:
              Service: fargate.amazonaws.com
            Action: kms:GenerateDataKeyWithoutPlaintext
            Condition:
              StringEquals:
                kms:EncryptionContext:aws:ecs:clusterAccount:
                  - !Ref AWS::AccountId
                kms:EncryptionContext:aws:ecs:clusterName:
                  - !Sub "${AWS::StackName}-ECS-Cluster" # circular dependency
            Resource: "*"
          - Sid: "ECS: Allow grant creation permission for Fargate tasks."
            Effect: Allow
            Principal:
              Service: fargate.amazonaws.com
            Action: kms:CreateGrant
            Condition:
              StringEquals:
                kms:EncryptionContext:aws:ecs:clusterAccount:
                  - !Ref AWS::AccountId
                kms:EncryptionContext:aws:ecs:clusterName:
                  - !Sub "${AWS::StackName}-ECS-Cluster" # circular dependency
              ForAllValues:StringEquals:
                kms:GrantOperations:
                  - Decrypt
            Resource: "*"
          - Sid: "CW Log Groups: Allow Log Groups to access KMS Key."
            Effect: Allow
            Principal:
              Service: !Sub "logs.${AWS::Region}.amazonaws.com"
            Action:
              - kms:Encrypt*
              - kms:Decrypt*
              - kms:ReEncrypt*
              - kms:GenerateDataKey*
              - kms:Describe*
            Resource: '*'
            Condition:
              ArnLike:
                kms:EncryptionContext:aws:logs:arn: !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*"


  NLB:
    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
    Properties:
      IpAddressType: ipv4
      Scheme: internal
      Type: network
      SubnetMappings:
        - SubnetId: <SubnetID>


  Listener:
    Type: AWS::ElasticLoadBalancingV2::Listener
    Properties:
      LoadBalancerArn: !GetAtt NLB.LoadBalancerArn
      Port: 8443
      Protocol: TCP
      DefaultActions:
        - Type: forward
          TargetGroupArn: !GetAtt TargetGroup.TargetGroupArn


  TargetGroup:
    Type: AWS::ElasticLoadBalancingV2::TargetGroup
    Properties:
      VpcId: <VPCID>
      Port: 8443
      Protocol: TCP
      TargetType: ip
      TargetGroupAttributes:
        - Key: preserve_client_ip.enabled
          Value: true
        - Key: deregistration_delay.connection_termination.enabled
          Value: true
      HealthCheckEnabled: true


  Cluster:
    Type: AWS::ECS::Cluster
    Properties:
      ClusterName: !Sub "${AWS::StackName}-ECS-Cluster"
      ClusterSettings:
        - Name: containerInsights
          Value: enabled
      CapacityProviders:
        - FARGATE
      Configuration:
        ManagedStorageConfiguration:
          FargateEphemeralStorageKmsKeyId: !GetAtt KMSKey.KeyId


  SecurityGroup:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: "Security Group"
      VpcId: <VPCID>
      Tags:
        - Key: Name
          Value: !Sub "${AWS::StackName}-ECS-SecurityGroup"


  Service:
    Type: AWS::ECS::Service
    DependsOn:
      - NLB
      - Listener
    Properties:
      Cluster: !Ref Cluster
      TaskDefinition: !GetAtt TaskDefinition.TaskDefinitionArn
      DesiredCount: 1
      LaunchType: FARGATE
      LoadBalancers:
        - ContainerName: "Test"
          ContainerPort: 8443
          TargetGroupArn: !GetAtt TargetGroup.TargetGroupArn
      PropagateTags: SERVICE
      NetworkConfiguration:
        AwsvpcConfiguration:
          SecurityGroups:
            - !GetAtt SecurityGroup.GroupId
          Subnets:
            - <Subnet>


  ECSTaskIAMRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: ecs-tasks.amazonaws.com
            Action: sts:AssumeRole


  ECSExecutionIAMRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: ecs-tasks.amazonaws.com
            Action: sts:AssumeRole
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy


  TaskDefinition:
    Type: AWS::ECS::TaskDefinition
    Properties:
      TaskRoleArn: !GetAtt ECSTaskIAMRole.Arn
      ExecutionRoleArn: !GetAtt ECSExecutionIAMRole.Arn
      NetworkMode: awsvpc
      RuntimePlatform:
        CpuArchitecture: ARM64
      Cpu: 256
      Memory: 512
      Volumes:
        - Name: ecs-exec-lib
        - Name: ecs-exec-log
      ContainerDefinitions:
        - Name: "Test"
          Image: <Image>
          ReadonlyRootFilesystem: true
          MountPoints:
            - SourceVolume: ecs-exec-lib
              ContainerPath: "/var/lib/amazon"
            - SourceVolume: ecs-exec-log
              ContainerPath: "/var/log/amazon"
          PortMappings:
            - ContainerPort: 8443 # TCP
          LogConfiguration:
            LogDriver: awslogs
            Options:
              awslogs-region: !Ref AWS::Region
              awslogs-group: !Ref LogGroup
              awslogs-stream-prefix: !Ref AWS::StackName


  LogGroup:
    Type: AWS::Logs::LogGroup
    Properties:
      KmsKeyId: !GetAtt KMSKey.Arn
      RetentionInDays: 30
```

**Comment by kddejong** [comment]
Thank you.  Yea this is absolutely the issue that we resolved in v1.9.1.  You should be good to go.
